tests/unit/conftest.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""pytest fixtures for the unit test."""
+
+import contextlib
+import typing
+import unittest.mock
+
+import ops
+import pytest
+from ops.model import BlockedStatus, Container
+from ops.testing import Harness
+
+from tests.unit import helpers
+from tests.unit._patched_charm import DiscourseCharm, pgsql_patch
+
+BLOCKED_STATUS = BlockedStatus.name  # type: ignore
+
+DATABASE_NAME = "discourse"
+
+
+@pytest.fixture(name="start_harness", autouse=True)
+def start_harness_fixture():  # -> typing.Generator[Harness, None, None]:
+    """Ops testing framework harness fixture."""
+
+    def start_harness(
+        *,
+        with_postgres: bool = True,
+        with_redis: bool = True,
+        with_ingress: bool = False,
+        with_config: typing.Optional[typing.Dict[str, typing.Any]] = None,
+    ):
+        """Start a harness discourse charm.
+
+        This is also a workaround for the fact that Harness
+        doesn't reinitialize the charm as expected.
+        Ref: https://github.com/canonical/operator/issues/736
+
+        Args:
+            - with_postgres: should a postgres relation be added
+            - with_redis: should a redis relation be added
+        """
+        harness = Harness(DiscourseCharm)
+        harness.disable_hooks()
+        # Protected access is needed to instantiate harness
```

Comment:
This seems to warrant a more detailed comment, I would be surprised if it is intentional to use the private harness methods to instantiate the harness

Reply:
Removed